backend/fms_core/template_importer/row_handlers/normalization/normalization.py

```diff
@@ -0,0 +1,92 @@
+            destination_container, _, self.errors['container'], self.warnings['container'] = get_or_create_container(
+                barcode=destination_container_dict['barcode'],
+                kind=destination_container_dict['kind'],
+                name=destination_container_dict['name'],
+                coordinates=destination_container_dict['coordinates'],
+                container_parent=container_parent_obj)
+
+            resulting_sample, self.errors['transfered_sample'], self.warnings['transfered_sample'] = transfer_sample(
+                process=process_measurement['process'],
+                sample_source=source_sample_obj,
+                container_destination=destination_container,
+                volume_used=process_measurement['volume_used'],
+                execution_date=process_measurement['execution_date'],
+                coordinates_destination=destination_sample['coordinates'],
+                volume_destination=destination_sample['volume'],
+                source_depleted=source_sample['depleted'],
+                comment=process_measurement['comment'])
+
+            concentration = None
+            # Case when ng/uL is given
+            if destination_sample['concentration_ngul']:
+                concentration = destination_sample['concentration_ngul']
+            # Case when nM is given
+            elif destination_sample['concentration_nm']:
+                if source_sample_obj.is_library:
+                    library = source_sample_obj.derived_sample_not_pool.library
+                    if library.library_size:
+                        concentration = convert_concentration_from_nm_to_ngbyul(destination_sample['concentration_nm'],
+                                                                                library.molecular_weight_approx,
+                                                                                library.library_size)
+                    else:
+                        self.errors['library'] = 'Library size has not been set for this library.'
+                    if concentration is None:
+                        self.errors['concentration'] = 'Failed to convert concentration from nM to ng/uL'
+                else:
+                    self.errors['concentration'] = 'Concentration specified in nM should be only for libraries.'
+
+            if concentration is not None:
+                _, self.errors['concentration_update'], self.warnings['concentration_update'] = \
+                    update_sample(sample_to_update=resulting_sample, concentration=concentration)
+
+            # Set the process measurement properties
+            process_measurement_properties['Final Volume']['value'] = destination_sample['volume']
+            process_measurement_properties['Final Concentration']['value'] = concentration
+
+            # Create process measurement's properties
+            process_measurement_obj = ProcessMeasurement.objects.get(source_sample=source_sample_obj)
```

Comment:
get() returned more than one ProcessMeasurement -- it returned 3!
This filter is not enough specific ...